SrfGen/NHM/GNS_ERF_comparison/gns_erf_comparison.py

```diff
@@ -0,0 +1,203 @@
+                dip,
+                direction,
+                depth_to_base,
+                depth_to_top,
+                end_1,
+                end_2,
+                mag,
+                rp,
+                line_segments,
+                trace_points
+            ),
+            line_ix + 4 + n_line_sections + 1,
+        )
+
+
+def compare(gns_fault: GNSFault, nhm_fault: NHMFault):
+    result_str, diff = f"\n{nhm_fault.name} has different:\n", False
+    if nhm_fault.dip != gns_fault.dip:
+        result_str += f"\t - Dip: GNS - {gns_fault.dip}, NHM - {nhm_fault.dip}\n"
+        diff = True
+    if not np.isclose(nhm_fault.dip_dir, gns_fault.dip_dir, rtol=0.01):
+        result_str += (
+            f"\t - Dip direction: GNS - {gns_fault.dip_dir}, NHM - {nhm_fault.dip_dir}\n"
+        )
+        diff = True
+    # if np.round(nhm_fault.mw, 1) != gns_fault.mag:
+    if not np.isclose(nhm_fault.mw, gns_fault.mag, 0.01):
+        result_str += f"\t - Magnitude: GNS - {gns_fault.mag}, NHM - {nhm_fault.mw}\n"
+        diff = True
+    # if (
+    #     (np.round(nhm_fault.recur_int_median, -1) != gns_fault.rp)
+    #     and (np.round(nhm_fault.recur_int_median, -2) != gns_fault.rp)
+    #     and (np.round(nhm_fault.recur_int_median, -3) != gns_fault.rp)
+    #     and (np.round(nhm_fault.recur_int_median, -4) != gns_fault.rp)
+    # ):
+    if not np.isclose(nhm_fault.recur_int_median, gns_fault.rp, rtol=0.05):
+        result_str += f"\t - Return period: GNS - {gns_fault.rp}, NHM - {nhm_fault.recur_int_median}\n"
+        diff = True
+    if nhm_fault.dtop != gns_fault.depth_to_top:
+        result_str += f"\t - Dtop: GNS - {gns_fault.depth_to_top}, NHM - {nhm_fault.dtop}\n"
+        diff = True
+    if nhm_fault.dbottom != gns_fault.depth_to_base:
+        result_str += f"\t - Dbottom: GNS - {gns_fault.depth_to_base}, NHM - {nhm_fault.dbottom}\n"
+        diff = True
+
+    # NHM contains points along the trace, whereas GNS is individual line segments
+    if gns_fault.trace.shape[0] != nhm_fault.trace.shape[0] or not np.all(np.isclose(nhm_fault.trace, gns_fault.trace)):
+        result_str += f"\t - Fault traces:\nGNS \n{gns_fault.trace}\nNHM \n{nhm_fault.trace}\n"
+        diff = True
+
+    # # Compare line segments
+    # for ix in range(nhm_fault.trace.shape[0]):
+    #     nhm_loc = nhm_fault.trace[ix]
+    #     if (ix + 1) < nhm_fault.trace.shape[0]:
+    #         gns_loc_1, gns_loc_2 = gns_fault.line_segments[ix]
+    #         gns_loc = np.asarray(gns_loc_1)
+    #     else:
+    #         gns_loc_1, gns_loc_2 = gns_fault.line_segments[ix - 1]
+    #         gns_loc = np.asarray(gns_loc_2)
+    #
+    #     if not np.all(np.isclose(nhm_loc, gns_loc)):
+    #         diff = True
+
+    if diff:
+        return result_str
+    return None
+
+
+if __name__ == "__main__":
+    standard_erf_ffp = (
+        os.path.joins(os.path.dirname(__file__), "../NZ_FLTmodel_2010_v18p6.txt")
+    )
+    gns_erf_ffp = "./F501111U.DAT"
```

Comment:
just a quick note that this may crash if CWD is does not contains this file, i.e. executing gns_erf_comparison.py from other places.
Otherwise everything else looks good.